zlib-rs and cloudflare_zlib features conflict

If you try to enable both the zlib-rs and cloudflare_zlib features:

```toml
[dependencies]
flate2 = { version = "1.0.35", features = ["zlib-rs", "cloudflare_zlib"] }
```
Then there are some compile errors:
```
error[E0252]: the name `libz` is defined multiple times
   --> /home/jonathan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flate2-1.0.35/src/ffi/c.rs:412:9
    |
409 |     use libz_rs_sys as libz;
    |         ------------------- previous import of the module `libz` here
...
412 |     use cloudflare_zlib_sys as libz;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `libz` reimported here
    |
    = note: `libz` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
    |
412 |     use cloudflare_zlib_sys as other_libz;
    |                             ~~~~~~~~~~~~~
...
```
I believe this is can be fixed by establishing precedence between the two flags here: 
https://github.com/rust-lang/flate2-rs/blob/14aec222af2b85485bd3df4578d8d510eca46af7/src/ffi/c.rs#L408-L412

--- a/src/ffi/c.rs
+++ b/src/ffi/c.rs
@@ -50,27 +50,38 @@ impl Default for StreamWrapper {
                 reserved: 0,
                 opaque: ptr::null_mut(),
                 state: ptr::null_mut(),
-                #[cfg(all(
-                    feature = "any_zlib",
-                    not(any(feature = "cloudflare-zlib-sys", feature = "libz-rs-sys"))
+
+                #[cfg(any(
+                    // zlib-ng
+                    feature = "zlib-ng",
+                    // libz-sys
+                    all(not(feature = "cloudflare_zlib"), not(feature = "zlib-ng"), not(feature = "zlib-rs"))
                 ))]
                 zalloc: allocator::zalloc,
-                #[cfg(all(
-                    feature = "any_zlib",
-                    not(any(feature = "cloudflare-zlib-sys", feature = "libz-rs-sys"))
+                #[cfg(any(
+                    // zlib-ng
+                    feature = "zlib-ng",
+                    // libz-sys
+                    all(not(feature = "cloudflare_zlib"), not(feature = "zlib-ng"), not(feature = "zlib-rs"))
                 ))]
                 zfree: allocator::zfree,
 
-                #[cfg(all(feature = "any_zlib", feature = "cloudflare-zlib-sys"))]
+                #[cfg(
+                    // cloudflare-zlib
+                    all(feature = "cloudflare_zlib", not(feature = "zlib-rs"), not(feature = "zlib-ng")),
+                )]
                 zalloc: Some(allocator::zalloc),
-                #[cfg(all(feature = "any_zlib", feature = "cloudflare-zlib-sys"))]
+                #[cfg(
+                    // cloudflare-zlib
+                    all(feature = "cloudflare_zlib", not(feature = "zlib-rs"), not(feature = "zlib-ng")),
+                )]
                 zfree: Some(allocator::zfree),
 
                 // for zlib-rs, it is most efficient to have it provide the allocator.
                 // The libz-rs-sys dependency is configured to use the rust system allocator
-                #[cfg(all(feature = "any_zlib", feature = "libz-rs-sys"))]
+                #[cfg(all(feature = "zlib-rs", not(feature = "zlib-ng")))]
                 zalloc: None,
-                #[cfg(all(feature = "any_zlib", feature = "libz-rs-sys"))]
+                #[cfg(all(feature = "zlib-rs", not(feature = "zlib-ng")))]
                 zfree: None,
             })),
         }
@@ -87,7 +98,14 @@ impl Drop for StreamWrapper {
     }
 }
 
-#[cfg(all(feature = "any_zlib", not(feature = "libz-rs-sys")))]
+#[cfg(any(
+    // zlib-ng
+    feature = "zlib-ng",
+    // cloudflare-zlib
+    all(feature = "cloudflare_zlib", not(feature = "zlib-rs"), not(feature = "zlib-ng")),
+    // libz-sys
+    all(not(feature = "cloudflare_zlib"), not(feature = "zlib-ng"), not(feature = "zlib-rs")),
+))]
 mod allocator {
     use super::*;
 
@@ -405,17 +423,19 @@ mod c_backend {
     #[cfg(feature = "zlib-ng")]
     use libz_ng_sys as libz;
 
-    #[cfg(all(not(feature = "zlib-ng"), feature = "zlib-rs"))]
+    #[cfg(all(feature = "zlib-rs", not(feature = "zlib-ng")))]
     use libz_rs_sys as libz;
 
-    #[cfg(all(not(feature = "zlib-ng"), feature = "cloudflare_zlib"))]
+    #[cfg(
+        // cloudflare-zlib
+        all(feature = "cloudflare_zlib", not(feature = "zlib-rs"), not(feature = "zlib-ng")),
+    )]
     use cloudflare_zlib_sys as libz;
 
-    #[cfg(all(
-        not(feature = "cloudflare_zlib"),
-        not(feature = "zlib-ng"),
-        not(feature = "zlib-rs")
-    ))]
+    #[cfg(
+        // libz-sys
+        all(not(feature = "cloudflare_zlib"), not(feature = "zlib-ng"), not(feature = "zlib-rs")),
+    )]
     use libz_sys as libz;
 
     pub use libz::deflate as mz_deflate;

